app/wasm/test/messages_test.go

```diff
@@ -0,0 +1,246 @@
+package wasm
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v7/app/wasm"
+	wasmbindings "github.com/osmosis-labs/osmosis/v7/app/wasm/bindings"
+	"github.com/stretchr/testify/assert"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSwap(t *testing.T) {
```

Comment:
This is cool test that doesn't involve calling the contract, but just the message handler function. Good one

Reply:
Good idea, I'll add a couple multi-hop checks here. Along with checking the different error conditions for them.